### PR TITLE
Switch to golangci-lint

### PR DESCRIPTION
**Pull request by saschagrunert**
This commit vendors golangci-lint and enables all linters which do not
complain right now. CI and scripting has been adapted as well.

### CODE REVIEW

**Comment by vrothberg** [comment]
Please use the module in `./tests/tools` for vendoring. This will avoid mixing tests and build dependencies in one module.

File: go.mod [hunk 1]
```diff
@@ -22,12 +21,10 @@ require (
 	github.com/fsouza/go-dockerclient v1.3.0
 	github.com/ghodss/yaml v1.0.0
 	github.com/gogo/protobuf v1.2.0 // indirect
+	github.com/golangci/golangci-lint v1.17.1
```
Comment (vrothberg) [inline go.mod]: This should be moved over ot https://github.com/containers/buildah/tree/master/tests/tools .

**Comment by saschagrunert** [comment]
> Please use the module in `./tests/tools` for vendoring. This will avoid mixing tests and build dependencies in one module.

Thanks for pointing me there, I was searching it but could not find it for whatever reasons. :innocent:

**Comment by vrothberg** [comment]
No worries :) I am happy that you're pulling the trigger on this migration. I opened an [issue for libpod](https://github.com/containers/libpod/issues/3498) today to track the migration there as well - still quite some work needed to let it pass though.

File: go.mod (hunk 1 above)
Comment (saschagrunert) [reply go.mod]: Done :+1:

**Comment by saschagrunert** [comment]
We have to install ostree on travis to make the linter work there. I will check if we can do that easily... :)

**Comment by saschagrunert** [comment]
Seems to work :+1:

**Comment by rhatdan** [comment]
LGTM
@rh-atomic-bot r+

**Pull request closed without merging**